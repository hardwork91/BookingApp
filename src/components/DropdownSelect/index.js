import React, { Component } from "react";
import PropTypes from "prop-types";
import { Icon, Menu, Dropdown, Tooltip } from "antd";
import { TOOLTIP_DELAY } from "../../Containers/BookingForm/constants";

export default class DropdownSelect extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: props.value,
    };
  }

  handleSelect = ({ key }) => {
    const { onChange, handleClearResults } = this.props;
    this.setState({ value: key }, () => {
      onChange(key);
      handleClearResults();
    });
  };

  renderMenu = () => (
    <Menu selectedKeys={this.state.value} onClick={this.handleSelect}>
      {this.props.options.map(option => (
        <Menu.Item key={option}>{option}</Menu.Item>
      ))}
    </Menu>
  );

  render() {
    const { disabled, title } = this.props;
    return (
      <Tooltip title={title} mouseEnterDelay={TOOLTIP_DELAY}>
        <Dropdown
          overlay={this.renderMenu()}
          trigger={["click"]}
          disabled={disabled}
        >
          <a className="ant-dropdown-link" onClick={e => e.preventDefault()}>
            {this.state.value} <Icon type="down" />
          </a>
        </Dropdown>
      </Tooltip>
    );
  }
}

DropdownSelect.propTypes = {
  value: PropTypes.string,
  options: PropTypes.array.isRequired,
  disabled: PropTypes.bool,
  title: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  handleClearResults: PropTypes.func.isRequired,
};
