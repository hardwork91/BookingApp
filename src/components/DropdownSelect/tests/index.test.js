import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { mount, configure } from "enzyme";
import toJson from "enzyme-to-json";
import DropdownSelect from "../index";

configure({ adapter: new Adapter() });

const setup = props => ({
  options: [{ key: "1" }],
  title: "Travel Class",
  handleClearResults: jest.fn(),
  ...props,
});

describe("DropdownSelect", () => {
  it("render as expected", () => {
    const tree = mount(<DropdownSelect {...setup()} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("when handleSelect", () => {
    const props = setup({ onChange: jest.fn() });
    const tree = mount(<DropdownSelect {...props} />);
    const key = "1";
    tree.instance().handleSelect({ key });
    expect(tree.state().value).toEqual(key);
    expect(props.onChange).toBeCalled();
    expect(props.handleClearResults).toBeCalled();
  });

  it("when handleOnClick", () => {
    const tree = mount(<DropdownSelect {...setup()} />);
    const preventDefault = jest.fn();
    tree.instance().handleOnClick({ preventDefault });
    expect(preventDefault).toBeCalled();
  });
});
