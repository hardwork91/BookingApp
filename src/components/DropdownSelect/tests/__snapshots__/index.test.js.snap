// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownSelect render as expected 1`] = `
<DropdownSelect
  handleClearResults={[Function]}
  options={
    Array [
      Object {
        "key": "1",
      },
    ]
  }
  title="Travel Class"
>
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    mouseEnterDelay={1}
    mouseLeaveDelay={0.1}
    placement="top"
    title="Travel Class"
    transitionName="zoom-big-fast"
  >
    <Tooltip
      align={Object {}}
      arrowContent={null}
      arrowPointAtCenter={false}
      autoAdjustOverflow={true}
      builtinPlacements={
        Object {
          "bottom": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tc",
              "bc",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "bottomLeft": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "bottomRight": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tr",
              "br",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "left": Object {
            "ignoreShake": true,
            "offset": Array [
              -4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "cr",
              "cl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "leftBottom": Object {
            "ignoreShake": true,
            "offset": Array [
              -4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "br",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "leftTop": Object {
            "ignoreShake": true,
            "offset": Array [
              -4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tr",
              "tl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "right": Object {
            "ignoreShake": true,
            "offset": Array [
              4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "cl",
              "cr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "rightBottom": Object {
            "ignoreShake": true,
            "offset": Array [
              4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bl",
              "br",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "rightTop": Object {
            "ignoreShake": true,
            "offset": Array [
              4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "tr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "top": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              -4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bc",
              "tc",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "topLeft": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              -4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bl",
              "tl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "topRight": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              -4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "br",
              "tr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
        }
      }
      destroyTooltipOnHide={false}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.1}
      onPopupAlign={[Function]}
      onVisibleChange={[Function]}
      overlay="Travel Class"
      placement="top"
      prefixCls="ant-tooltip"
      title="Travel Class"
      transitionName="zoom-big-fast"
      trigger={
        Array [
          "hover",
        ]
      }
      visible={false}
    >
      <Trigger
        action={
          Array [
            "hover",
          ]
        }
        afterPopupVisibleChange={[Function]}
        arrowContent={null}
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        blurDelay={0.15}
        builtinPlacements={
          Object {
            "bottom": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tc",
                "bc",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "bottomLeft": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tl",
                "bl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "bottomRight": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tr",
                "br",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "left": Object {
              "ignoreShake": true,
              "offset": Array [
                -4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "cr",
                "cl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "leftBottom": Object {
              "ignoreShake": true,
              "offset": Array [
                -4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "br",
                "bl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "leftTop": Object {
              "ignoreShake": true,
              "offset": Array [
                -4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tr",
                "tl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "right": Object {
              "ignoreShake": true,
              "offset": Array [
                4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "cl",
                "cr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "rightBottom": Object {
              "ignoreShake": true,
              "offset": Array [
                4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bl",
                "br",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "rightTop": Object {
              "ignoreShake": true,
              "offset": Array [
                4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tl",
                "tr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "top": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                -4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bc",
                "tc",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "topLeft": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                -4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bl",
                "tl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "topRight": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                -4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "br",
                "tr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
          }
        }
        defaultPopupVisible={false}
        destroyPopupOnHide={false}
        focusDelay={0}
        getDocument={[Function]}
        getPopupClassNameFromAlign={[Function]}
        hideAction={Array []}
        mask={false}
        maskClosable={true}
        mouseEnterDelay={1}
        mouseLeaveDelay={0.1}
        onPopupAlign={[Function]}
        onPopupVisibleChange={[Function]}
        overlay="Travel Class"
        popup={[Function]}
        popupAlign={Object {}}
        popupClassName=""
        popupPlacement="top"
        popupStyle={Object {}}
        popupTransitionName="zoom-big-fast"
        popupVisible={false}
        prefixCls="ant-tooltip"
        showAction={Array []}
        title="Travel Class"
        visible={false}
      >
        <Dropdown
          key="trigger"
          mouseEnterDelay={0.15}
          mouseLeaveDelay={0.1}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          overlay={
            <Menu
              onClick={[Function]}
              selectedKeys={undefined}
            >
              <MenuItem>
                Object {
                  "key": "1",
                }
              </MenuItem>
            </Menu>
          }
          placement="bottomLeft"
          trigger={
            Array [
              "click",
            ]
          }
        >
          <Dropdown
            defaultVisible={false}
            mouseEnterDelay={0.15}
            mouseLeaveDelay={0.1}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onVisibleChange={[Function]}
            overlay={[Function]}
            overlayClassName=""
            overlayStyle={Object {}}
            placement="bottomLeft"
            prefixCls="ant-dropdown"
            showAction={Array []}
            transitionName="slide-up"
            trigger={
              Array [
                "click",
              ]
            }
          >
            <Trigger
              action={
                Array [
                  "click",
                ]
              }
              afterPopupVisibleChange={[Function]}
              blurDelay={0.15}
              builtinPlacements={
                Object {
                  "bottomCenter": Object {
                    "offset": Array [
                      0,
                      4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "tc",
                      "bc",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "bottomLeft": Object {
                    "offset": Array [
                      0,
                      4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "tl",
                      "bl",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "bottomRight": Object {
                    "offset": Array [
                      0,
                      4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "tr",
                      "br",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "topCenter": Object {
                    "offset": Array [
                      0,
                      -4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "bc",
                      "tc",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "topLeft": Object {
                    "offset": Array [
                      0,
                      -4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "bl",
                      "tl",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "topRight": Object {
                    "offset": Array [
                      0,
                      -4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "br",
                      "tr",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                }
              }
              defaultPopupVisible={false}
              defaultVisible={false}
              destroyPopupOnHide={false}
              focusDelay={0}
              getDocument={[Function]}
              getPopupClassNameFromAlign={[Function]}
              hideAction={Array []}
              mask={false}
              maskClosable={true}
              mouseEnterDelay={0.15}
              mouseLeaveDelay={0.1}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onPopupAlign={[Function]}
              onPopupVisibleChange={[Function]}
              onVisibleChange={[Function]}
              overlay={[Function]}
              popup={[Function]}
              popupAlign={Object {}}
              popupClassName=""
              popupPlacement="bottomLeft"
              popupStyle={Object {}}
              popupTransitionName="slide-up"
              popupVisible={false}
              prefixCls="ant-dropdown"
              showAction={Array []}
            >
              <a
                className="ant-dropdown-link ant-dropdown-trigger"
                key="trigger"
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
              >
                 
                <Icon
                  type="down"
                >
                  <LocaleReceiver
                    componentName="Icon"
                  >
                    <i
                      aria-label="icon: down"
                      className="anticon anticon-down"
                    >
                      <IconReact
                        className=""
                        type="down-o"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-down"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            key="svg-down-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </i>
                  </LocaleReceiver>
                </Icon>
              </a>
            </Trigger>
          </Dropdown>
        </Dropdown>
      </Trigger>
    </Tooltip>
  </Tooltip>
</DropdownSelect>
`;
